### PR TITLE
Add a generic client that can be used for 3rd party objects.

### PR DESCRIPTION
**Pull request by brendandburns**
@krousey @lavalamp

### CODE REVIEW

**Comment by krousey** [comment]
Going to assign this to @caesarxuchao for now since I'm out for a while, but a couple of comments.

If this is meant to be the start of dynamic client, consider the package structure proposed in https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/client-package-structure.md#package-structure. Also consider using [unstructure schema](https://github.com/kubernetes/kubernetes/blob/e309583ff116ff6ec6adc7c70999c8c9b77fd64b/pkg/runtime/unstructured.go) instead of just returning raw bytes.

**Comment by brendandburns** [comment]
Nope, this isn't intended to be the beginning of a dynamic client, its intended to simply be a usable client for opaque types.

I don't see the win from using the unstructured codec, since at the end of the day, you just have bytes anyway...

--brendan

File: pkg/client/unversioned/extensions.go [hunk 1]
```diff
@@ -94,6 +96,10 @@ func (c *ExtensionsClient) ThirdPartyResources(namespace string) ThirdPartyResou
 	return newThirdPartyResources(c, namespace)
 }
 
+func (c *ExtensionsClient) Generic(namespace string, gvk unversioned.GroupVersionKind) GenericResourceInterface {
+	return newGenericResources(c, namespace, gvk)
```
Comment (caesarxuchao) [inline pkg/client/unversioned/extensions.go]: This doesn't really belong to the ExtensionsClient. IMO, it should be parallel to the dynamic client krousey mentioned.

For now, I think it makes more sense to create a GenericClient, parallel to ExtensionsClient and Client, and it should be constructed with a RESTClient. WDYT?

Comment (lavalamp) [inline pkg/client/unversioned/extensions.go]: Yeah, this doesn't belong in the extensions client. Is there a reason you can't just use the DynamicClient (other than that it doesn't exist yet?), if not maybe just create that? It looks like you're 85% there already.

Comment (caesarxuchao) [inline pkg/client/unversioned/extensions.go]: cc @nikhiljindal

**Comment by caesarxuchao** [comment]
Is this PR still needed? Can you use the dynamic client instead (https://github.com/kubernetes/kubernetes/blob/master/pkg/client/typed/dynamic/client.go)?

**Comment by fejta** [comment]
This PR has had no meaningful activity for multiple months. If it is still valid please rebase, push a new commit and reopen the PR. Thanks!

**Pull request closed without merging**